src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -167,6 +184,32 @@ private AnalyzerExecutor(
             _addSuppression = addSuppression;
 
             _compilationAnalysisValueProviderFactory = new CompilationAnalysisValueProviderFactory();
+
+            if (getAnalyzerConfigOptionsProvider != null)
+            {
+                var hasDifferentOptions = false;
+
+                var map = new Dictionary<DiagnosticAnalyzer, AnalyzerOptions>(ReferenceEqualityComparer.Instance);
```

Comment:
Do you want to set the initial capacity here? I can imagine you might decide not to (to avoid the large allocation), but I'd be generally betting that those that passed the function for the per-diagnostic analyzers knows they're going to need it?